ephios/core/views/event.py

```diff
@@ -539,23 +536,25 @@ class EventCopyView(CustomPermissionRequiredMixin, SingleObjectMixin, FormView):
     permission_required = "core.add_event"
     model = Event
     template_name = "core/event_copy.html"
-    form_class = EventDuplicationForm
+    form_class = EventCopyForm
 
     def setup(self, request, *args, **kwargs):
         super().setup(request, *args, **kwargs)
         self.object = self.get_object()
 
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs["original_start"] = timezone.localtime(
+            self.object.shifts.first().start_time
+        ).isoformat()
+        return kwargs
+
+    def get_success_url(self):
+        return reverse("core:event_copy", kwargs={"pk": self.object.pk})
+
     def form_valid(self, form):
-        tz = get_current_timezone()
-        occurrences = form.cleaned_data["recurrence"].between(
-            datetime.now() - timedelta(days=1),
-            datetime.now() + timedelta(days=7305),  # allow dates up to twenty years in the future
-            inc=True,
-            dtstart=datetime.combine(
-                forms.DateField().to_python(self.request.POST["start_date"]), datetime.min.time()
-            ),
-        )
-        for date in occurrences:
+        tz = timezone.get_current_timezone()
+        for date in form.cleaned_data["recurrence"].xafter(datetime.now(), 1000, inc=True):
```

Comment:
Seems like I can preview dates in the past but events/shifts in the past wont be created! Either subtract a century from datetime.now here or disallow past dates in the frontend